src/Cookie.php

```diff
@@ -4,148 +4,150 @@
+		return $this->expires;
+	}
 
-    public function setExpires(int $expires): void {
-        $this->expires = $expires;
-    }
+	public function setExpires(int $expires): void {
+		$this->expires = $expires;
+	}
 
-    public function getPath(): string {
-        return $this->path;
-    }
+	public function isExpired(): bool {
+		return $this->expires < time();
+	}
 
-    public function setPath(string $path): void {
-        $this->path = $path;
-    }
+	public function getRemainingTime(): int {
+		return $this->expires - \time();
+	}
 
-    public function getDomain(): string {
-        return $this->domain;
-    }
+	public function getPath(): string {
+		return $this->path;
+	}
 
-    public function setDomain(string $domain): void {
-        $this->domain = $domain;
-    }
+	public function setPath(string $path): void {
+		$this->path = $path;
+	}
 
-    public function getSecure(): bool {
-        return $this->secure;
-    }
+	public function getDomain(): string {
+		return $this->domain;
+	}
 
-    public function setSecure(bool $secure): void {
-        $this->secure = $secure;
-    }
+	public function setDomain(string $domain): void {
+		$this->domain = $domain;
+	}
 
-    public function getHttpOnly(): bool {
-        return $this->httpOnly;
-    }
+	public function isSecure(): bool {
+		return $this->secure;
+	}
+
+	public function setSecure(bool $secure): void {
+		$this->secure = $secure;
+	}
+
+	public function getHttpOnly(): bool {
+		return $this->httpOnly;
+	}
 
-    public function setHttpOnly(bool $httpOnly): void {
-        $this->httpOnly = $httpOnly;
-    }
+	public function setHttpOnly(bool $httpOnly): void {
+		$this->httpOnly = $httpOnly;
+	}
 
-    public function getOptions(): array {
-        return $this->options;
-    }
+	public function getOptions(): array {
+		return $this->options;
+	}
 
-    public function setOptions(array $options): void {
-        $this->options = $options;
-    }
+	public function setOptions(array $options): void {
+		$this->options = $options;
+	}
 
 	/**
-	 * @return mixed|null
+	 * @return self
 	 */
-    public static function getCookie(string $name) {
-        if (!\array_key_exists($name, $_COOKIE)) {
-            return null;
-        }
-
-        return $_COOKIE[$name];
-    }
-
-    public static function deleteCookie(string $name): bool {
-        if (!\array_key_exists($name, $_COOKIE)) {
-            return false;
-        }
-
-		unset($_COOKIE[$name]);
-
-		return true;
-    }
-
-    public static function saveCookie(Cookie $cookie): bool {
-		return setcookie(
-			$cookie->name,
-			$cookie->value,
-			$cookie->expires,
-			$cookie->path,
-			$cookie->domain,
-			$cookie->secure,
-			$cookie->httpOnly,
-		);
-    }
+	public static function load(string $name) {
+		if (!\array_key_exists($name, $_COOKIE)) {
+			return null;
+		}
 
-    public static function cookieExists(string $name): bool {
-        return \array_key_exists($name, $_COOKIE);
-    }
+		//return $_COOKIE[$name];
```

Comment:
Instead of return the cookie value this should return the a object of Cookie class. Otherwise it make not much sense to manage cookie with this class.